### PR TITLE
Upgrade graphics to LF 6.6.23-2.0.0

### PR DESCRIPTION
**Pull request by valijec**
Upgrade graphics to LF 6.6.23-2.0.0:
- weston: Upgrade from 12.0.3.imx to 12.0.4.imx
- wayland-protocols: Replace hardcoded SRC_URI with WAYLAND_PROTOCOLS_SRC variable
- mesa-demos: Add patch for egl clear backgrounds black color
- mesa: Add patches for video showing wrong frame and fix virgl driver assert issue
- imx-gpu-viv: Upgrade from 6.4.11.p2.4-* to 6.4.11.p2.6-*
- imx-dpu-g2d: Upgrade imxdpu-g2d from 2.1.12 to 2.1.14
- imx-gpu-g2d: Upgrade from 6.4.11.p2.4 to 6.4.11.p2.6
- imx-g2d-samples: Update to latest 431f311a
- imx-pxp-g2d: Update to latest edd234c4e

Fix issue #1861

### CODE REVIEW

**Comment by MaxKrummenacher** [comment]
Hi
Thanks for the patches.

I think the change to the `wayland imx` should go into its own recipe. The change is unrelated to the `weston imx` update.

Additionally I would reword the weston patch to make it clear it is about the imx fork, e.g. `weston: Upgrade from 12.0.3.imx to 12.0.4.imx`

**Comment by MaxKrummenacher** [comment]
Looks good to me.

**Comment by valijec** [comment]
Thank you @MaxKrummenacher I rebased the branch to include latest commits from [Freescale:master](https://github.com/Freescale/meta-freescale/tree/master)

**Comment by angolini** [comment]
I think this merge should wait the linux-imx bump. Other than that I think it's ok

File: recipes-graphics/imx-g2d/imx-gpu-g2d_6.4.11.p2.6.bb [hunk 1]
```diff
@@ -1,21 +1,21 @@
 # Copyright (C) 2016 Freescale Semiconductor
-# Copyright 2017-2024 NXP
+# Copyright 2017-2022 NXP
```
Comment (otavio) [inline recipes-graphics/imx-g2d/imx-gpu-g2d_6.4.11.p2.6.bb:2]: Keep old date.

**Review by otavio — CHANGES_REQUESTED** [review]

File: recipes-graphics/imx-g2d/imx-dpu-g2d_2.1.14.bb [hunk 2]
```diff
@@ -36,10 +36,14 @@ do_install () {
     cp -Pr ${S}/g2d/usr/include/* ${D}${includedir}
 }
 
+INSANE_SKIP:append:libc-musl = " file-rdeps"
+RDEPENDS:${PN}:append:libc-musl = " gcompat"
+
```
Comment (thochstein) [inline recipes-graphics/imx-g2d/imx-dpu-g2d_2.1.14.bb:41]: Good, it looks like I accidentally removed this in the last update.

**Review by thochstein — APPROVED** [review]

**Comment by otavio** [comment]
Good, just squash the change in the previous commit and push it again to the branch, so I can merge it.

**Comment by valijec** [comment]
> Good, just squash the change in the previous commit and push it again to the branch, so I can merge it.

Done, sorry for that, now should be ok
